### ISSUE TITLE
Macro for I2C_MODE or SPI_MODE clashes with other libraries - use enum instead

### ISSUE THREAD

**Issue by euphi** [sparkfun/SparkFun_BME280_Arduino_Library#54]
SparkFunBME280.h uses many macro definition where enum would be a cleaner solution. Especially very common names like I2C_MODE or SPI_MODE are problematic.

https://github.com/sparkfun/SparkFun_BME280_Arduino_Library/blob/c42e78089d9cda457d616715b41c970e253c2acf/src/SparkFunBME280.h#L46
https://github.com/sparkfun/SparkFun_BME280_Arduino_Library/blob/c42e78089d9cda457d616715b41c970e253c2acf/src/SparkFunBME280.h#L47

E. g. when installing the BMI160 library there are tons of subsequent error message just due to these _rogue_ macro definitions. 

Cleanest solution (see patch below) is to move them into the class (and thus the class' namespace) and define them as anonymous enum values.

> In file included from src/I2CSensors.h:11,
>                  from src/I2CSensors.cpp:10:
> .pio/libdeps/trgb-esp32-s3/SparkFun BME280/src/SparkFunBME280.h:46:18: error: expected identifier before numeric constant
>  #define I2C_MODE 0
>                   ^
> .pio/libdeps/trgb-esp32-s3/BMI160-Arduino/BMI160Gen.h:9:43: note: in expansion of macro 'I2C_MODE'
>          typedef enum { INVALID_MODE = -1, I2C_MODE = 1, SPI_MODE = 2 } Mode;
>                                            ^~~~~~~~
> .pio/libdeps/trgb-esp32-s3/SparkFun BME280/src/SparkFunBME280.h:46:18: error: expected '}' before numeric constant
>  #define I2C_MODE 0
>                   ^
> .pio/libdeps/trgb-esp32-s3/BMI160-Arduino/BMI160Gen.h:9:43: note: in expansion of macro 'I2C_MODE'
>          typedef enum { INVALID_MODE = -1, I2C_MODE = 1, SPI_MODE = 2 } Mode;
>                                            ^~~~~~~~
> In file included from src/I2CSensors.h:14,
>                  from src/I2CSensors.cpp:10:
> .pio/libdeps/trgb-esp32-s3/BMI160-Arduino/BMI160Gen.h:9:22: note: to match this '{'
>          typedef enum { INVALID_MODE = -1, I2C_MODE = 1, SPI_MODE = 2 } Mode;
>                       ^
> In file included from src/I2CSensors.h:11,
>                  from src/I2CSensors.cpp:10:
> .pio/libdeps/trgb-esp32-s3/SparkFun BME280/src/SparkFunBME280.h:46:18: error: expected unqualified-id before numeric constant
>  #define I2C_MODE 0
>                   ^
> .pio/libdeps/trgb-esp32-s3/BMI160-Arduino/BMI160Gen.h:9:43: note: in expansion of macro 'I2C_MODE'
>          typedef enum { INVALID_MODE = -1, I2C_MODE = 1, SPI_MODE = 2 } Mode;
>                                            ^~~~~~~~
> In file included from src/I2CSensors.h:14,
>                  from src/I2CSensors.cpp:10:
> .pio/libdeps/trgb-esp32-s3/BMI160-Arduino/BMI160Gen.h:11:20: error: 'bool begin' redeclared as different kind of symbol
>          bool begin(Mode mode, const int arg1 = 0x68, const int arg2 = 2);
>                     ^~~~

Patch:

```C++
--- ".pio/libdeps/trgb-esp32-s3/SparkFun BME280/src/SparkFunBME280.h"   2024-12-02 22:08:47.334115549 +0100
+++ ".pio/libdeps/trgb-esp32-s3-FL/SparkFun BME280/src/SparkFunBME280.h"        2020-12-31 04:44:27.000000000 +0100
@@ -43,6 +43,9 @@
 //You will need to have the SoftwareWire library installed
 //#include <SoftwareWire.h> //SoftwareWire by Testato. Installed from library manager.
 
+#define I2C_MODE 0
+#define SPI_MODE 1
+
 #ifndef BME280_SPI_CLOCK
 #ifdef ARDUINO_ARCH_ESP32
 #define BME280_SPI_CLOCK 1000000
@@ -183,12 +186,6 @@
 class BME280
 {
   public:
-
-    enum {
-       I2C_MODE=0,
-       SPI_MODE=1
-    };
-          
     //settings
     BME280_SensorSettings settings;
        SensorCalibration calibration;
```

This is an incompatible change, you need to explicitly stating the class name when using it *outside* the BME280 class:

```C++
void I2CSensors::initBME280() {
        bme280.settings.commInterface = BME280::I2C_MODE;
        bme280.settings.I2CAddress = 0x76;
...
```

**Comment by edspark** [comment]
Agreed, I decided to stick with the macro style that is already there but changed the name to something more in line with our [style guidlines](https://github.com/sparkfun/.github/blob/main/SFECodeStyleGuide.md).
